src/main/java/pl/betoncraft/betonquest/Journal.java

```diff
@@ -249,18 +250,25 @@ private String generateMainPage() {
                     // check conditions and continue loop if not met
                     final String rawConditions = section.getString(key + ".conditions");
                     if (rawConditions != null && rawConditions.length() > 0) {
-                        for (final String condition : rawConditions.split(",")) {
-                            try {
-                                final ConditionID conditionID = new ConditionID(pack, condition);
-                                if (!BetonQuest.condition(playerID, conditionID)) {
-                                    continue keys;
-                                }
-                            } catch (ObjectNotFoundException e) {
-                                LogUtils.getLogger().log(Level.WARNING, "Error while generatin main page in " + PlayerConverter.getPlayer(playerID)
-                                        + "'s journal - condition '" + condition + "' not found: " + e.getMessage());
-                                LogUtils.logThrowable(e);
-                                continue keys;
+                        try {
+                            final Collection<ConditionID> pageConditions = Arrays.stream(rawConditions.split(","))
+                                    .map(con -> {
+                                        try {
+                                            return new ConditionID(pack, con);
+                                        } catch (ObjectNotFoundException e) {
+                                            throw new IllegalArgumentException("Error while generating main page in " + PlayerConverter.getPlayer(playerID)
+                                                    + "'s journal - condition '" + con + "' not found: " + e.getMessage(), e);
+                                        }
+                                    })
+                                    .collect(Collectors.toList());
+
+                            if (!BetonQuest.condition(playerID, pageConditions)) {
+                                continue;
                             }
+                        } catch (IllegalArgumentException exception) {
+                            LogUtils.getLogger().log(Level.WARNING, exception.getMessage());
+                            LogUtils.logThrowable(exception);
+                            continue;
                         }
```

Comment:
Parallelization not required for creating conditions as the checking is done later (and also parallelized).
```suggestion
                        try {
                            final List<ConditionID> pageConditions = new ArrayList<>();
                            for (String conditionString : rawConditions.split(",")) {
                                pageConditions.add(new ConditionID(pack, conditionString));
                            }

                            if (!BetonQuest.condition(playerID, pageConditions)) {
                                continue;
                            }
                        } catch (ObjectNotFoundException exception) {
                            LogUtils.getLogger().log(Level.WARNING, "Error while generating main page in " + PlayerConverter.getPlayer(playerID) + "'s journal - condition '" + con + "' not found: " + e.getMessage());
                            LogUtils.logThrowable(exception);
                            continue;
                        }
```